common/app/targeting/CampaignAgent.scala

```diff
@@ -0,0 +1,27 @@
+package targeting
+
+import common._
+import com.gu.targeting.client.CampaignCache
+import conf.Configuration
+import scala.util.control.NonFatal
+import scala.concurrent.Future
+
+object CampaignAgent extends Logging with ExecutionContexts {
+  private val agent = AkkaAgent[CampaignCache](CampaignCache(List()))
+
+  def refresh(): Future[Unit] = {
+    Configuration.targeting.campaignsUrl.map(url => {
+      CampaignCache.fetch(url).map(agent.send)
+    }).getOrElse(Future.failed(throw new BadConfigurationException("Campaigns URL not configured")))
```

Comment:
The `fetch` returns a `Future[CampaignCache]` which I map to `Future[Unit]` by sending it to the akka agent. The scheduler wants a `Future[Unit]` so this is convenient.

Thanks for the notice about the throw, totally a schoolboy error on my part 😞